File: Coding_Problems/7_reverseNumber.js
function reverseNumber(n) {
  let reversed = 0;
  let num = Math.abs(n);

  while (num > 0) {
    reversed = reversed * 10 + (num % 10);
    num = Math.floor(num / 10);
  }

  return n < 0 ? -reversed : reversed;
}

File: Coding_Problems/CPP/7_reverseNumber.cpp
#include <iostream>
#include <cmath>

using namespace std;

int reverseNumber(int n)
{
    int reversed = 0;
    int num = abs(n);

    while (num > 0)
    {
        reversed = reversed * 10 + (num % 10);
        num /= 10;
    }

    return (n < 0) ? -reversed : reversed;
}

int main()
{
    cout << reverseNumber(12345) << endl;
    cout << reverseNumber(-9876);

    return 0;
}